Transport/BacnetIpUdpProtocolTransport.cs

```diff
@@ -371,8 +371,8 @@ private static UnicastIPAddressInformation GetAddressDefaultInterface()
             .Where(a => a.Address.AddressFamily == AddressFamily.InterNetwork)
             .ToArray();
 
-        return unicastAddresses.Length == 1
-            ? unicastAddresses.Single()
+        return unicastAddresses.Length >= 1
+            ? unicastAddresses.First()
             : null;

```

Comment:
How about we remove `.ToArray()` and do 

```csharp
return unicastAddresses.FirstOrDefault()
```

Reply:
But, this would change the original behavior. What is the issue issue with having `null` returned here? Is it the invalid broadcast address used later?

Reply:
Exactly, when null is returned the broadcast address is resolved to 255.255.255.255 which is not allowed in BACnet.

Reply:
Taking one step back, the original problem is when using 0.0.0.0 when you have multipe interfaces active or an interface in use with multiple ip's bound to it. -> In this case, it's not clear which broadcast address should be used at all?
In my opinion, there has to be a check when initializing the udp transport to bind to a dedicated ip address available on the device, if multiple options are available. In this case, the correct broadcast address will always be resolved automatically!

Reply:
That is correct! 0.0.0.0 will listen on all interfaces -> the resolution of the broadcast address is the only issue here.
Regarding the validity of the 255.255.255.255 broadcast address in BACnet, I'll have a detailed look on the spec.
As far as I know it is not allowed. Unfortunately, I already saw devices from big vendors relying on it, though. 
-> has to be double checked before merging :-(

Will give an update on this...

Reply:
> A UDP broadcast packet is 'sent' to an IP address with the subnet of the broadcasting device and all 1's in the host portion. For example, if a device has an address 128.253.109.10 and a subnet mask 255.255.255.0, it can use the broadcast address 128.253.109.255. Although most networks also allow the use of the IP broadcast address 255.255.255.255, we have ruled this out for BACnet/IP because of some IP protocol stack limitations that certain implementors have encountered.

src: http://www.bacnet.org/Tutorial/BACnetIP/sld006.html

Reply:
This bug bit us a few times. I would prefer some sort of a thrown exception if a reasonable interface can't be found (after filtering out `lo`). Using first available interface may not be good, as it would be non-deterministic, and might pick the wrong one

Reply:
+1 this bit me the first time I tested the library out as well. 

I would prefer an exception if no suitable interface could be found. If multiple suitable interfaces are found, either pick the first one or better yet - being able to specify which one based on some sort of rule set (interface name, type, identifier etc). Or expose a public method that gets the list of suitable interfaces and accept one of  them as the argument.